pkg/scheduler/internal/heap/heap.go

```diff
@@ -115,7 +114,7 @@ func (h *data) Pop() interface{} {
 }
 
 // Peek is supposed to be called by heap.Peek only.
-func (h *data) Peek() interface{} {
+func (h *data[T]) Peek() interface{} {
```

Comment:
why not return `T` instead of `interface{}`?

Reply:
Good catch. `Peek()` can return `T` because it is not backed by [`container/heap`](https://pkg.go.dev/container/heap#Interface). Also, changed the comment not to be misleading.